docs/contributing/testing-guidelines/integration-testing.md

```diff
@@ -0,0 +1,293 @@
+
+### Integration test with a single node: component test
+
+The simplest scenario is a single node. In this case, the integration test is commonly referred to as a component test.
+
+To add a component test to an existing node, follow the example of the `lanelet2_map_provider` package that has an executable named `lanelet2_map_provider_exe`.
+
+In `package.xml`, add
+
+```xml
+<test_depend>ros_testing</test_depend>
+```
+
+In `CMakeLists.txt`, add or modify the `BUILD_TESTING` section:
+
+```cmake
+if(BUILD_TESTING)
+  add_ros_test(
+    test/lanelet2_map_provider_launch.test.py
+    TIMEOUT "30"
+  )
+endif()
+```
+
+The `TIMEOUT` argument is given in seconds; see [here](https://github.com/ros2/ros_testing/blob/master/ros_testing/cmake/add_ros_test.cmake) for details.
+
+To create test follow [launch_testing quick-start example](https://github.com/ros2/launch/tree/master/launch_testing#quick-start-example).
+
+Let's look at `test/lanelet2_map_provider_launch.test.py` as an example.
+
+The essential content is to first import dependencies:
+
+```python
+from ament_index_python import get_package_share_directory
+from launch import LaunchDescription
+from launch_ros.actions import Node
+import launch_testing
+
+import os
+import pytest
+import unittest
+```
+
+Then a launch description was created to launch the node under test. Note that `test_map.osm` file path is found and passed to the node. This is one of limitation of [smoke test](#integration-testing-smoke-test):
+
+```python
+@pytest.mark.launch_test
+def generate_test_description():
+
+    map_osm_file = os.path.join(
+        get_package_share_directory('lanelet2_map_provider'),
+        'data/test_map.osm'
+    )
+
+    lanelet2_map_provider = Node(
+        package='lanelet2_map_provider',
+        executable='lanelet2_map_provider_exe',
+        namespace='had_maps',
+        parameters=[
+            os.path.join(
+                get_package_share_directory('lanelet2_map_provider'),
+                'param/test.param.yaml'
+            ),
+            {
+                'map_osm_file': map_osm_file
+            }]
+    )
+
+    context = {'lanelet2_map_provider': lanelet2_map_provider}
+
+    return LaunchDescription([
+        lanelet2_map_provider,
+        # Start tests right away - no need to wait for anything
+        launch_testing.actions.ReadyToTest()]
+    ), context
+```
+
+and finally the test condition. As before, it is just a smoke test ensures the node can be
+
+1. launched with its default parameter file,
+2. terminated with a standard `SIGTERM` signal,
+
+so the test code is executed after the node executable has been shut down (`post_shutdown_test`):
+
+```python
+@launch_testing.post_shutdown_test()
+class TestProcessOutput(unittest.TestCase):
+
+    def test_exit_code(self, proc_output, proc_info, ndt_mapper):
+        # Check that process exits with code 0
+        launch_testing.asserts.assertExitCodes(proc_info, process=ndt_mapper)
+```
+
+## Running the test
```

Comment:
Let's move the descriptions about how to run tests to another page.
We need to keep pages short and simple.